templates/workloads/partials/cf/subscribe.yml

```diff
@@ -0,0 +1,176 @@
+  Type: AWS::SNS::Subscription
+  Properties:
+    TopicArn: !Join ['', [!Sub 'arn:${AWS::Partition}:sns:${AWS::Region}:${AWS::AccountId}:', !Ref AppName, '-', !Ref EnvName, '-{{logicalIDSafe $topic.Service}}-{{logicalIDSafe $topic.Name}}']]
+    Protocol: 'sqs'
+    {{- if $topic.Queue}}
+    Endpoint: !GetAtt {{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}EventsQueue.Arn
+    {{- else}}
+    Endpoint: !GetAtt EventsQueue.Arn
+    {{- end}}
+
+{{- if $topic.Queue}}
+{{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}EventsQueue:
+  Type: AWS::SQS::Queue
+  Properties:
+    KmsMasterKeyId: 'alias/aws/sqs'
+    {{- if $topic.Queue.Retention}}
+    MessageRetentionPeriod: {{$topic.Queue.Retention}}
+    {{- end}}
+    {{- if $topic.Queue.Delay}}
+    DelaySeconds: {{$topic.Queue.Delay}}
+    {{- end}}
+    {{- if $topic.Queue.Timeout}}
+    VisibilityTimeout: {{$topic.Queue.Timeout}}
+    {{- end}}
+    {{- if $topic.Queue.FIFO}}
+    FifoQueue: true
+    {{- if $topic.Queue.FIFO.HighThroughput}}
+    FifoThroughputLimit: 'perMessageGroupId'
+    DeduplicationScope: 'messageGroup'
+    {{- end}}
+    {{- end}}
+    {{- if $topic.Queue.DeadLetter}}
+    RedrivePolicy:
+      deadLetterTargetArn: !GetAtt {{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}DeadLetterQueue.Arn
+      maxReceiveCount: {{$topic.Queue.DeadLetter.Tries}}
+    {{- end}}
+
+{{- if $topic.Queue.DeadLetter}}
+{{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}DeadLetterQueue:
+  Type: AWS::SQS::Queue
+
+{{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}DeadLetterPolicy:
+  Type: AWS::SQS::QueuePolicy
+  Properties:
+    Queues: [!Ref '{{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}DeadLetterQueue']
+    PolicyDocument:
+      Version: "2012-10-17"
+      Statement:
+        - Sid: Allow-TaskRole-Subscribe
+          Effect: Allow
+          Principal:
+            Service: sns.amazonaws.com
+          Action: 
+            - sqs:GetMessage
+            - sqs:DeleteMessage
+          Resource: !GetAtt {{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}EventsQueue.Arn
+          Condition:
+            ArnEquals:
+              aws:SourceArn: !Ref 'TaskRole'
+{{- end}}
+
+{{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}QueuePolicy:
+  Type: AWS::SQS::QueuePolicy
+  Properties:
+    Queues: [!Ref '{{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}EventsQueue']
+    PolicyDocument:
+      Version: '2012-10-17'
+      Statement:
+        - Sid: Allow-TaskRole-Subscribe
+          Effect: Allow
+          Principal:
+            Service: sns.amazonaws.com
+          Action: 
+            - sqs:GetMessage
+            - sqs:DeleteMessage
+          Resource: !GetAtt {{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}EventsQueue.Arn
+          Condition:
+            ArnEquals:
+              aws:SourceArn: !Ref 'TaskRole'
```

Comment:
Wait do we need to update all the queue policies such that the Principal is the task role for the first element?

```
          Sid: Allow-TaskRole-Subscribe
          Effect: Allow
          Principal:
            AWS:
            - !Ref TaskRole
          Action: 
            - sqs:GetMessage
            - sqs:DeleteMessage
            - sqs:DeleteMessageBatch
          Resource: !GetAtt {{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}EventsQueue.Arn
```